### DISCUSSION TITLE
Exportar valores de inputs entre distintos componentes - VUEX

### DISCUSSION THREAD

**Discussion by ide45** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#569] [Q&A] [answered]
¿Cómo le hago? ¿Voy por buen camino? Se ocupan promesas? 

![imagen](https://user-images.githubusercontent.com/91546625/161175312-0ae4916d-fcd0-4d7e-bec3-75996060cd6e.png)

![imagen](https://user-images.githubusercontent.com/91546625/161175146-10ba873d-7a94-4ed8-b622-f7e1655c2120.png)

Qué fuentes me recomiendan investigar?

Gracias, de antemano.

**Comment by doguedogue** [comment]
Creo que asi como lo tines es con Vue y con el state es Vuex.

Te recomiendo el curso de Fernado Herrera: Vue.js: De cero a experto
o algo parecido si te quieres especializar y/o entenderle super bien, aunque con las prácticas y programas vistos aqui se puede sacar el proyecto.

**Comment by richirrim** [comment]
Broh para obtener los valores de los inputs lo ideal y lo correcto sería hacerlo usando la directiva `v-model`. En cada `input` coloca la directiva y le pasas el nombre de la variable que creaste en tu modelo `data() { ... }`. 

Ejemplo:
```js
// En vez de ir directamente 
// al DOM usando getElementById(). 
// Coloca en cada input el v-model.

<input v-model="name" />

// Y en tu data solo lo dejas así.
data() { 
    return {
        name: ' '
    }
}

```
Y haces lo mismo con los demás inputs.
Y para exportar datos a otros componentes necesitas usar eventos personalizados.  Y así pasar datos del componente hijo al padre.

**Comment by ChristianDeM** [comment]
Como te comentaron lo ideal seria hacerlo con Vuex ya que desde el store es mas facil mover las variables de cada input  a diferentes componentes y views.

**Comment by ide45** [comment, marked as answer]
Muchas gracias a todos los que respondieron, supe qué buscar y se pudo resolver por sus comentarios. Al final lo hice de esta manera.

STATE
![imagen](https://user-images.githubusercontent.com/91546625/161357793-2b8d59e2-54c3-4111-aaf7-e0fd02c1b0a0.png)

MUTATION
![imagen](https://user-images.githubusercontent.com/91546625/161357835-850ba75a-79f5-4668-b154-0c2edabace70.png)

Componente de origen
![imagen](https://user-images.githubusercontent.com/91546625/161357894-7fe26490-c389-4c15-ae4d-46d4eb73d8f0.png)

![imagen](https://user-images.githubusercontent.com/91546625/161357901-358bfe0e-536c-4888-b439-ef1f70247ea1.png)


Componente de destino
![imagen](https://user-images.githubusercontent.com/91546625/161357875-e2017551-c9d9-4146-ab77-5b3cc16c71e1.png)

**Reply by Jose-Galicia** [reply]
Yo también tenia esa duda, estaba usando los v-model para hacer eso, pero voy a hacerlo de la misma manera.
